MiddlesexMA/src/main/java/com/openlattice/middlesex/MiddlesexCharges2.java

```diff
@@ -98,29 +102,38 @@ public static void main( String[] args ) throws InterruptedException {
                 .option( "header", "true" )
                 .load( arrestsPath );
 
-        Dataset<Row> chargeData = sparkSession
+        Dataset<Row> incident16Data = sparkSession
                 .read()
                 .format( "com.databricks.spark.csv" )
                 .option( "header", "true" )
-                .load( chargesPath );
-
-        RequiredEdmElements requiredEdmElements = ConfigurationService.StaticLoader
-                .loadConfiguration( RequiredEdmElements.class );
-        FullQualifedNameJacksonDeserializer.registerWithMapper( ObjectMappers.getYamlMapper() );
-        FullQualifedNameJacksonDeserializer.registerWithMapper( ObjectMappers.getJsonMapper() );
-        if ( requiredEdmElements != null ) {
-            RequiredEdmElementsManager reem = new RequiredEdmElementsManager( edm,
-                    retrofit.create( PermissionsApi.class ) );
-            reem.ensureEdmElementsExist( requiredEdmElements );
-        }
+                .load( incidents2016Path );
+
+        Dataset<Row> incident17Data = sparkSession
+                .read()
+                .format( "com.databricks.spark.csv" )
+                .option( "header", "true" )
+                .load( incidents2017Path );
+
+//        RequiredEdmElements requiredEdmElements = ConfigurationService.StaticLoader
+//                .loadConfiguration( RequiredEdmElements.class );
+//        FullQualifedNameJacksonDeserializer.registerWithMapper( ObjectMappers.getYamlMapper() );
+//        FullQualifedNameJacksonDeserializer.registerWithMapper( ObjectMappers.getJsonMapper() );
+//        if ( requiredEdmElements != null ) {
+//            RequiredEdmElementsManager reem = new RequiredEdmElementsManager( edm,
+//                    retrofit.create( PermissionsApi.class ) );
+//            reem.ensureEdmElementsExist( requiredEdmElements );
+//        }
 
         Flight flight = Flight.newFlight()
                 .createEntities()
 
-                .addEntity( "suspect" )
+                .addEntity( "arrestee" )
                 .to( "LPDArrestSuspects" )
                 .addProperty( new FullQualifiedName( "nc.SubjectIdentification" ) )
-                .value( MiddlesexCharges2::getSubjectIdentification ).ok()
+                .value(row -> {
+                    return Parsers.getAsString(row.getAs("Name")) + " " + Parsers.getAsString(row.getAs("DOB"));
```

Comment:
is name+DOB guaranteed to be unique for each person?

Reply:
I think it is the most unique combination given the limited data there is here - before the person ID was the booking number, which I confirmed in the csvs was definitely not unique - high utilizers appeared in multiple bookings, and this was making the top utilizer results erroneous. Basically, the top utilizer was always only in 1 arrest, which was wrong. Here I checked the first few top utilizers in Excel by hand, and it matched what is now in gallery.

Reply:
That's the problem linking is supposed to fix -- unless Middlesex has told use that name+dob is safe to use as a unique combo, we should probably just assign each person a random id and let linking decide whether or not to merge people

Reply:
I will change it - although it seems like pre-linking, it is working perfectly right now.